### ISSUE TITLE
Conformer encoder layer as decoder in Transducer ESPNETv1

### ISSUE THREAD

**Issue by nguyenhuy1209** [espnet/espnet#3797]
Hi, I am trying to use a Conformer Encoder as a Transducer Decoder in the VIVOS recipe:

```
# minibatch related
batch-size: 32r
maxlen-in: 800
maxlen-out: 150

# optimization related
criterion: loss
early-stop-criterion: "validation/main/loss"
sortagrad: 0
opt: noam
noam-lr: 2.0
noam-adim: 320
optimizer-warmup-steps: 4000
epochs: 40
patience: 0
accum-grad: 2
grad-clip: 5.0

# network architecture
## general
custom-enc-positional-encoding-type: rel_pos
custom-enc-self-attn-type: rel_self_attn
## encoder related
etype: custom
custom-enc-input-layer: vgg2l
enc-block-arch:
        - type: conformer
          d_hidden: 320
          d_ff: 320
          heads: 4
          macaron_style: True
          use_conv_mod: True
          conv_mod_kernel: 15
          dropout-rate: 0.3
          att-dropout-rate: 0.3
          pos-dropout-rate: 0.3
enc-block-repeat: 8

## decoder related
dtype: custom
custom-dec-input-layer: embed
dec-block-arch:
        - type: conformer
          d_hidden: 320
          d_ff: 320
          heads: 4
          macaron_style: True
          use_conv_mod: True
          conv_mod_kernel: 15
          dropout-rate: 0.3
          att-dropout-rate: 0.3
          pos-dropout-rate: 0.3
dec-block-repeat: 4
## joint network related
joint-dim: 128

# transducer related
model-module: "espnet.nets.pytorch_backend.e2e_asr_transducer:E2E"
```

As I read the code, if we use conformer for the decoder type, the build_blocks() function in transducer/blocks.py will create a ConformerEncoderLayer instance for the transducer decoder. The model was able to train smoothly, but the loss went down rapidly in a weird way (in the figure). When I use the last epoch (epoch 16) to run decoding, the model outputs nothing but empty sequence "". Can anyone explain this phenomenon? Is this a bug in the transducer's loss function, or is it impractical to use Conformer Encoder as Transducer Decoder? Thanks.

![image](https://user-images.githubusercontent.com/47093934/142130728-523206f4-97a6-41c9-8fc9-632861cbec18.png)

![image](https://user-images.githubusercontent.com/47093934/142130550-2a2e0415-df51-439a-88f6-0cf7b17ff36e.png)

**Comment by b-flo** [comment]
Hi,

> Can anyone explain this phenomenon? Is this a bug in the transducer's loss function, or is it impractical to use Conformer Encoder as Transducer Decoder? Thanks.

Conformer decoder is not supported (or at least we don't have a definition for ConformerDecoderLayer). I forgot to add back a check when refactoring custom model building.

**Comment by nguyenhuy1209** [comment]
Hi, @b-flo , thank you for your reply. I still have a question on why can't we use a Conformer as a decoder, is it because of technical issues or because it is theoretically impractical?

**Comment by b-flo** [comment]
>  is it because of technical issues or because it is theoretically impractical?

It was only intended as an encoder for ASR models when it was added. Conformer could be used as decoder for Transducer model but some things need to be modified. I'll check that when I have time.
